recipes/fmt/all/test_cmakedeps/conanfile.py

```diff
@@ -0,0 +1,25 @@
+import os
+from conans import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps
+from conan.tools.cross_building import cross_building as tools_cross_building
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeDeps"
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["FMT_HEADER_ONLY"] = self.dependencies["fmt"].options.header_only
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join("..", "test_package"))
+        cmake.build()
+
+    def test(self):
+        if not tools_cross_building(self):
+            folder = self.settings.build_type if self.settings.os == "Windows" else ""
```

Comment:
seems fragile and generator dependent rather than os dependent.

Reply:
Absolutely. Right now I'm more interested in knowing about the differences between v1 generators and these v2 ones, than on something final.

Probably the proper way to go in this `conanfile.py`, would be to pass some `CMAKE_RUNTIME_OUTPUT_DIRECTORY` to the build here, so it is always the same regardless of the generator (toolchain) and operating system

Reply:
to be honest, I doubt the condition is correct. it may use a different "layout" depending on the generator (e.g. Visual Studio vs NMake vs Ninja). so it may break for some users'